lib/spack/spack/cmd/find.py

```diff
@@ -21,23 +22,18 @@
 
 def setup_parser(subparser):
     format_group = subparser.add_mutually_exclusive_group()
-    format_group.add_argument('-s', '--short',
-                              action='store_const',
-                              dest='mode',
-                              const='short',
-                              default='short',
-                              help='show only specs (default)')
-    format_group.add_argument('-p', '--paths',
-                              action='store_const',
-                              dest='mode',
-                              const='paths',
-                              help='show paths to package install directories')
     format_group.add_argument(
-        '-d', '--deps',
-        action='store_const',
-        dest='mode',
-        const='deps',
-        help='show full dependency DAG of installed packages')
+        "--format", action="store", default=None,
+        help="output specs with the specified format string")
+    format_group.add_argument(
+        "--json", action="store_true", default=False,
+        help="output specs as machine-readable json records")
+
+    subparser.add_argument('-d', '--deps', action='store_true',
+                           help='output dependencies along with found specs')
+
+    subparser.add_argument('-p', '--paths', action='store_true',
+                           help='show paths to package install directories')
 
     arguments.add_common_arguments(
         subparser, ['long', 'very_long', 'tags'])
```

Comment:
`long` and `very_long` should be made mutually exclusive, as it doesn't make any sense to use them together.

Reply:
it's hard to do this with the common arguments -- let's hold off on that.